<ui version="4.0" >
 <author></author>
 <comment></comment>
 <exportmacro></exportmacro>
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>637</width>
    <height>566</height>
   </rect>
  </property>
  <property name="sizePolicy" >
   <sizepolicy>
    <hsizetype>0</hsizetype>
    <vsizetype>0</vsizetype>
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="font" >
   <font>
    <family>Lucida Grande</family>
    <pointsize>13</pointsize>
    <weight>50</weight>
    <italic>false</italic>
    <bold>false</bold>
    <underline>false</underline>
    <strikeout>false</strikeout>
   </font>
  </property>
  <property name="windowTitle" >
   <string>Edit Constraints</string>
  </property>
  <layout class="QVBoxLayout" >
   <property name="margin" >
    <number>9</number>
   </property>
   <property name="spacing" >
    <number>6</number>
   </property>
   <item>
    <widget class="QGroupBox" name="groupBox" >
     <property name="title" >
      <string>Constraint</string>
     </property>
     <layout class="QGridLayout" >
      <property name="margin" >
       <number>9</number>
      </property>
      <property name="spacing" >
       <number>6</number>
      </property>
      <item row="0" column="5" >
       <widget class="QLabel" name="label_6" >
        <property name="text" >
         <string>Value</string>
        </property>
       </widget>
      </item>
      <item row="1" column="4" >
       <widget class="QSpinBox" name="atom4" >
        <property name="maximum" >
         <number>999</number>
        </property>
        <property name="minimum" >
         <number>1</number>
        </property>
        <property name="value" >
         <number>4</number>
        </property>
       </widget>
      </item>
      <item row="1" column="5" >
       <widget class="QDoubleSpinBox" name="constraintValue" >
        <property name="wrapping" >
         <bool>false</bool>
        </property>
        <property name="decimals" >
         <number>3</number>
        </property>
        <property name="maximum" >
         <double>180</double>
        </property>
        <property name="minimum" >
         <double>-180</double>
        </property>
        <property name="singleStep" >
         <double>0.001</double>
        </property>
       </widget>
      </item>
      <item row="0" column="1" >
       <widget class="QLabel" name="label_2" >
        <property name="text" >
         <string>Atom A</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3" >
       <widget class="QLabel" name="label_4" >
        <property name="text" >
         <string>Atom C</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1" >
       <widget class="QSpinBox" name="atom1" >
        <property name="maximum" >
         <number>999</number>
        </property>
        <property name="minimum" >
         <number>1</number>
        </property>
       </widget>
      </item>
      <item row="0" column="0" >
       <widget class="QLabel" name="label" >
        <property name="text" >
         <string>Type:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" >
       <widget class="QComboBox" name="constraintType" >
        <item>
         <property name="text" >
          <string>Stretch</string>
         </property>
        </item>
        <item>
         <property name="text" >
          <string>Bend</string>
         </property>
        </item>
        <item>
         <property name="text" >
          <string>Out of plane</string>
         </property>
        </item>
        <item>
         <property name="text" >
          <string>Dihedral</string>
         </property>
        </item>
        <item>
         <property name="text" >
          <string>Co-planar</string>
         </property>
        </item>
        <item>
         <property name="text" >
          <string>Perpendicular</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="7" >
       <widget class="QToolButton" name="addConstraint" >
        <property name="toolTip" >
         <string>Add constraint</string>
        </property>
        <property name="text" >
         <string>+</string>
        </property>
        <property name="icon" >
         <iconset>resources/icons/plus.png</iconset>
        </property>
       </widget>
      </item>
      <item row="1" column="2" >
       <widget class="QSpinBox" name="atom2" >
        <property name="maximum" >
         <number>999</number>
        </property>
        <property name="minimum" >
         <number>1</number>
        </property>
        <property name="value" >
         <number>2</number>
        </property>
       </widget>
      </item>
      <item row="0" column="4" >
       <widget class="QLabel" name="label_5" >
        <property name="text" >
         <string>Atom D</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2" >
       <widget class="QLabel" name="label_3" >
        <property name="text" >
         <string>Atom B</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3" >
       <widget class="QSpinBox" name="atom3" >
        <property name="maximum" >
         <number>999</number>
        </property>
        <property name="minimum" >
         <number>1</number>
        </property>
        <property name="value" >
         <number>3</number>
        </property>
       </widget>
      </item>
      <item row="1" column="6" >
       <spacer>
        <property name="orientation" >
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" >
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" >
     <property name="margin" >
      <number>0</number>
     </property>
     <property name="spacing" >
      <number>6</number>
     </property>
     <item>
      <widget class="QGroupBox" name="groupBox_3" >
       <property name="title" >
        <string>Dummy Atom:</string>
       </property>
       <layout class="QVBoxLayout" >
        <property name="margin" >
         <number>9</number>
        </property>
        <property name="spacing" >
         <number>6</number>
        </property>
        <item>
         <layout class="QHBoxLayout" >
          <property name="margin" >
           <number>0</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item>
           <widget class="QLabel" name="label_8" >
            <property name="text" >
             <string>Type:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="dummyType" >
            <item>
             <property name="text" >
              <string>Plane Normal</string>
             </property>
            </item>
            <item>
             <property name="text" >
              <string>Angle Bisector</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer>
            <property name="orientation" >
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" >
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" >
          <property name="margin" >
           <number>0</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item>
           <widget class="QLabel" name="label_9" >
            <property name="text" >
             <string>Atom A</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="dAtom1" >
            <property name="minimum" >
             <number>1</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_10" >
            <property name="text" >
             <string>B</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="dAtom2" >
            <property name="minimum" >
             <number>1</number>
            </property>
            <property name="value" >
             <number>2</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_11" >
            <property name="text" >
             <string>C</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="dAtom3" >
            <property name="minimum" >
             <number>1</number>
            </property>
            <property name="value" >
             <number>3</number>
            </property>
           </widget>
          </item>
          <item>
           <spacer>
            <property name="orientation" >
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" >
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QToolButton" name="addDummyAtom" >
            <property name="font" >
             <font>
              <family>Lucida Grande</family>
              <pointsize>11</pointsize>
              <weight>50</weight>
              <italic>false</italic>
              <bold>false</bold>
              <underline>false</underline>
              <strikeout>false</strikeout>
             </font>
            </property>
            <property name="toolTip" >
             <string>Add constraint</string>
            </property>
            <property name="text" >
             <string>+</string>
            </property>
            <property name="icon" >
             <iconset>resources/icons/plus.png</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_2" >
       <property name="title" >
        <string>Fixed Atom:</string>
       </property>
       <layout class="QVBoxLayout" >
        <property name="margin" >
         <number>9</number>
        </property>
        <property name="spacing" >
         <number>6</number>
        </property>
        <item>
         <layout class="QHBoxLayout" >
          <property name="margin" >
           <number>0</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item>
           <widget class="QLabel" name="label_7" >
            <property name="text" >
             <string>Atom Number:</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer>
            <property name="orientation" >
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" >
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QSpinBox" name="fixedAtom" >
            <property name="minimum" >
             <number>1</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" >
          <property name="margin" >
           <number>0</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item>
           <widget class="QLabel" name="label_14" >
            <property name="text" >
             <string>Fix</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="fixX" >
            <property name="text" >
             <string>X</string>
            </property>
            <property name="checked" >
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="fixY" >
            <property name="text" >
             <string>Y</string>
            </property>
            <property name="checked" >
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="fixZ" >
            <property name="text" >
             <string>Z</string>
            </property>
            <property name="checked" >
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer>
            <property name="orientation" >
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" >
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QToolButton" name="addFixedAtom" >
            <property name="toolTip" >
             <string>Add constraint</string>
            </property>
            <property name="text" >
             <string>+</string>
            </property>
            <property name="icon" >
             <iconset>resources/icons/plus.png</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_4" >
     <property name="title" >
      <string>Add Connectivity</string>
     </property>
     <layout class="QHBoxLayout" >
      <property name="margin" >
       <number>9</number>
      </property>
      <property name="spacing" >
       <number>6</number>
      </property>
      <item>
       <widget class="QLabel" name="label_13" >
        <property name="text" >
         <string>Atom</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="targetAtom" />
      </item>
      <item>
       <widget class="QLabel" name="label_12" >
        <property name="text" >
         <string>Connect To Atoms</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="cAtom1" />
      </item>
      <item>
       <widget class="QSpinBox" name="cAtom2" />
      </item>
      <item>
       <widget class="QSpinBox" name="cAtom3" />
      </item>
      <item>
       <widget class="QSpinBox" name="cAtom4" />
      </item>
      <item>
       <spacer>
        <property name="orientation" >
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" >
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QToolButton" name="addConnectivityButton" >
        <property name="toolTip" >
         <string>Add constraint</string>
        </property>
        <property name="text" >
         <string>+</string>
        </property>
        <property name="icon" >
         <iconset>resources/icons/plus.png</iconset>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QTableWidget" name="constraintTable" >
     <property name="horizontalScrollBarPolicy" >
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="editTriggers" >
      <set>QAbstractItemView::NoEditTriggers</set>
     </property>
     <property name="alternatingRowColors" >
      <bool>true</bool>
     </property>
     <property name="selectionMode" >
      <enum>QAbstractItemView::SingleSelection</enum>
     </property>
     <property name="selectionBehavior" >
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <property name="showGrid" >
      <bool>false</bool>
     </property>
     <property name="gridStyle" >
      <enum>Qt::NoPen</enum>
     </property>
     <property name="sortingEnabled" >
      <bool>false</bool>
     </property>
     <column>
      <property name="text" >
       <string>ID</string>
      </property>
     </column>
     <column>
      <property name="text" >
       <string>Type</string>
      </property>
     </column>
     <column>
      <property name="text" >
       <string>Atom 1</string>
      </property>
     </column>
     <column>
      <property name="text" >
       <string>Atom 2</string>
      </property>
     </column>
     <column>
      <property name="text" >
       <string>Atom 3</string>
      </property>
     </column>
     <column>
      <property name="text" >
       <string>Atom 4</string>
      </property>
     </column>
     <column>
      <property name="text" >
       <string>Value</string>
      </property>
     </column>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" >
     <property name="margin" >
      <number>0</number>
     </property>
     <property name="spacing" >
      <number>6</number>
     </property>
     <item>
      <spacer>
       <property name="orientation" >
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" >
        <size>
         <width>131</width>
         <height>31</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="deleteButton" >
       <property name="text" >
        <string>Delete</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="okButton" >
       <property name="text" >
        <string>OK</string>
       </property>
       <property name="default" >
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="cancelButton" >
       <property name="text" >
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <pixmapfunction></pixmapfunction>
 <tabstops>
  <tabstop>constraintType</tabstop>
  <tabstop>atom1</tabstop>
  <tabstop>atom2</tabstop>
  <tabstop>atom3</tabstop>
  <tabstop>atom4</tabstop>
  <tabstop>constraintValue</tabstop>
  <tabstop>addConstraint</tabstop>
  <tabstop>fixedAtom</tabstop>
  <tabstop>fixX</tabstop>
  <tabstop>fixY</tabstop>
  <tabstop>fixZ</tabstop>
  <tabstop>addFixedAtom</tabstop>
  <tabstop>dummyType</tabstop>
  <tabstop>dAtom1</tabstop>
  <tabstop>dAtom2</tabstop>
  <tabstop>dAtom3</tabstop>
  <tabstop>addDummyAtom</tabstop>
  <tabstop>constraintTable</tabstop>
  <tabstop>deleteButton</tabstop>
  <tabstop>okButton</tabstop>
  <tabstop>cancelButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>okButton</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>278</x>
     <y>253</y>
    </hint>
    <hint type="destinationlabel" >
     <x>96</x>
     <y>254</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cancelButton</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>369</x>
     <y>253</y>
    </hint>
    <hint type="destinationlabel" >
     <x>179</x>
     <y>282</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
